src/ploigos_step_runner/step_implementers/generate_evidence/generate_evidence.py

```diff
@@ -0,0 +1,237 @@
+                )
+                step_result.add_artifact(
+                    name='results-evidence-upload-results',
+                    description='Results of uploading the evidence JSON file ' \
+                        'to the given destination.',
+                    value=upload_result
+                )
+                step_result.add_artifact(
+                name='evidence-uri',
+                description='URI of the uploaded results archive.',
+                value=evidence_destination_uri
+                )
+            except RuntimeError as error:
+                step_result.success = False
+                step_result.message = str(error)
+                return step_result
+
+        step_result.success = True
+        step_result.message = 'Evidence successfully packaged ' \
+            'in JSON file and uploaded to data store.'
+
+        step_result.add_artifact(
+            name='result-generate-evidence-path',
+            value=evidence_path,
+            description='File path of evidence.'
+        )
+
+        return step_result
+
+    def __gather_evidence(self):
+        """Gather all collected evidence
+
+        Returns
+        -------
+        dict
+            Dictionary containing evidence (values) for each step (key) that
+            has collected evidence.
+
+        """
+
+        gathered_evidence = None
+        workflow_dict = {}
+        evidence_key = 'evidence'
+        piece_of_evidence_name = 'name'
+        attestations_keyword = 'attestations'
+
+        #Iterate over all previous step results
+        for previous_step_result in self.workflow_result.workflow_list:
+            step_name = previous_step_result.step_name
+
+            step_results = previous_step_result.get_step_result_dict()[step_name]
+            sub_steps = {}
+
+            #Iterate over sub steps in step result
+            for sub_step in step_results:
+                if step_results[sub_step][evidence_key]:
+                    evidence_dict = {}
+                    for piece_of_evidence in step_results[sub_step][evidence_key]:
+                        evidence_dict_key = piece_of_evidence[piece_of_evidence_name]
+                        evidence_dict[evidence_dict_key] = piece_of_evidence
+
+                    sub_step_evidence = {
+                        sub_step: {
+                            attestations_keyword:
+                                evidence_dict
+                        }
+                    }
+                    sub_steps = sub_step_evidence
+
+            #Add step to workflow dictionary
+            if sub_steps:
+                workflow_dict[step_name] = sub_steps
+                gathered_evidence = {
+                'apiVersion': GenerateEvidence.API_VERSION,
+                'kind': GenerateEvidence.KIND,
+                'workflow': workflow_dict
+            }
+
+        return gathered_evidence
+
+    def __write_evidence_to_json_file(self, evidence, evidence_path):
```

Comment:
what is this for?

Reply:
Oh that was for testing. I need to remove it...

Reply:
Removed.